spec/features/omniauth_spec.rb

```diff
@@ -118,9 +118,37 @@
     end
   end
 
+  shared_examples 'sign up with openid provider' do |provider|
+    scenario "#{provider} has option to log in" do
+      visit '/accounts/sign_in'
+      expect(page.has_content?('or sign in using')).to eq true
+      expect(page.has_link?("omniauth-#{provider}")).to eq true
+    end
+
+    scenario "#{provider} signs up new user" do
+      expected_count_openid = Openid.count + 1
+      expected_count_user = User.count + 1
+      visit '/accounts/sign_in'
+
+      mock_auth_accounts
+      within('#openidlinks') do
+        click_link "omniauth-#{provider}"
+      end
+      expect(flash).to eq("user-#{provider}@example.com signed in successfully with #{provider}")
+      expect(Openid.count).to eq(expected_count_openid)
+      expect(User.count).to eq(expected_count_user)
+    end
+  end
+
   describe 'omniauth' do
     if User.omniauth_providers.present?
       it_behaves_like 'sign in with openid'
     end
   end
+
+  describe 'omniauth provider' do
+    User.omniauth_providers.each do |provider|
```

Comment:
This can be moved in the previous describe 'omniauth' and inside the if statement.